room-gacha/index.ts

```diff
@@ -0,0 +1,218 @@
+import scrapeIt from 'scrape-it';
+import { promises as fs } from 'fs';
+import * as qs from 'querystring';
+import _ from 'lodash';
+import type { SlackInterface } from '../lib/slack';
+import { Prefectures, PrefectureKanji } from './Prefectures';
+
+interface Image {
+    url: string;
+    description: string;
+}
+
+interface Room {
+    name: string;
+    url: string;
+    images: Image[];
+    layoutImageUrl: string;
+    rent: string;
+    layout: string;
+    size: string;
+    direction: string;
+    years: string;
+    address: string;
+    access: string[];
+}
+
+const pickOneResult = async (cityIDs: string[], ar: string, bs: string, ta: string) => {
+    const appAddress = 'https://suumo.jp/jj/chintai/ichiran/FR301FC005/'; // 部屋ごとに表示
+    const queries = {
+        sc: cityIDs,
+        ar, bs, ta, // 謎 (必須) (都道府県固有)
+        po01: '09', // 並び替え: 新着順
+        pc: '100',  // 表示件数: 100件
+    };
+    const url = `${appAddress}?${qs.stringify(queries)}`;
+    interface SearchResult {
+        title: string;
+        hit: string;
+        rooms: Room[];
+    }
+    const result = await scrapeIt<SearchResult>(url, {
+        title: {
+            selector: '.ui-section-header h1',
+            convert: s => s.split(' ')[0],
+        },
+        hit: '.paginate_set-hit',
+        rooms: {
+            listItem: '.property',
+            data: {
+                name: '.property_inner-title a',
+                url: {
+                    selector: '.property_inner-title a',
+                    attr: 'href',
+                    convert: s => `https://suumo.jp${s}`,
```

Comment:
[comment] 文字列結合が若干気持ち悪いので、
```suggestion
                    convert: s => new URL(s, 'https://suumo.jp').href,
```
と書くこともできます

Reply:
おおーかしこい